### PR TITLE
Backend/allow missing categories

### PR DESCRIPTION
**Pull request by JaneBellaiche**
[ch-65149]

### CODE REVIEW

**Review by alexcombessie — APPROVED** [review]

File: python-lib/nlp/ontology_tagging/ontology_tagger.py [hunk 1]
```diff
@@ -141,14 +143,22 @@ def _set_log_level(self, languages: List[AnyStr]) -> None:
...
-        if self.ontology_df.empty:
+        self.ontology_df.dropna(
+            subset=[self.keyword_column, self.tag_column], inplace=True
+        )
+        if self.ontology_df[[self.keyword_column, self.tag_column]].empty:
             raise ValueError(
                 "No valid tags were found. Please specify at least a keyword and a tag in the ontology dataset, and re-run the recipe"
             )
 
+    def _replace_missing_categories(self) -> None:
```
Comment (alexcombessie) [inline python-lib/nlp/ontology_tagging/ontology_tagger.py:156]: Are we sure it's the right behavior to enforce?
Could you give an example of output in that case?

Comment (JaneBellaiche) [reply python-lib/nlp/ontology_tagging/ontology_tagger.py:156]: As I see it, there are two ways to handle missing categories:
- replace all missing categories with the same "default" one, but it is hard to find the right name for it 
- replace all missing categories with their tags, so the tags are the categories, but it creates non necessary categories.
Krishna have voted for the second option, but it is open to discussion 

Here is how it looks like 
Capture d’écran 2021-06-08 à 19 37 35
Capture d’écran 2021-06-08 à 19 41 26

Comment (alexcombessie) [reply python-lib/nlp/ontology_tagging/ontology_tagger.py:156]: I understand it's difficult to find the name for the "default" category.
However, in your screenshot, isn't it a bit weird to have a column named "politics" that will always only say "politics"? And same in the JSON, its says "politics": ["politics"]. If I were to try thinking like a new user, I would think it is a bug. 
Do you see what I mean?
If we were to put all cases without category in a single "uncategorized" category, would it look better?

Comment (JaneBellaiche) [reply python-lib/nlp/ontology_tagging/ontology_tagger.py:156]: That's a good idea, thank you 👍 fc93052

**Pull request merged**